File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-fixed/02-adversarial-non-openai-close.ts
import WebSocket from 'ws';

// Adversarial: a different WebSocket's close handler only logs, with no
// reconnect attempt — but it's not the OpenAI Realtime connection, so this
// rule should not flag it.
function setup(logger: any) {
  const internalSocket = new WebSocket('wss://internal.example.com/events');

  internalSocket.on('close', () => {
    logger.info('internal socket closed');
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-fixed/02_adversarial_non_openai_close.py
import logging

import websockets
from websockets.exceptions import ConnectionClosed

# Adversarial: a log-only close handler on a non-Realtime socket.
OTHER_URL = "wss://example.com/v1/stream"
logger = logging.getLogger(__name__)


async def stream():
    try:
        async with websockets.connect(OTHER_URL) as ws:
            async for message in ws:
                handle_message(message)
    except ConnectionClosed as exc:
        logger.error("socket closed: %s", exc)
